### ISSUE TITLE
cloud coverage calculation shouldn't return values larger than 100

### ISSUE THREAD

**Issue by scisco** [developmentseed/landsat-util#163]
Example scene: LC81180372016026LGN00

```

[34m===> [0mImage processing started for bands 4-3-2
    return self._write_to_file(new_bands, **rasterio_options)
[34m===> [0mGetting boundaries
  File "/usr/local/lib/python2.7/dist-packages/landsat/decorators.py", line 10, in wrapped_f
    return func(*args, **kwargs)
[34m===> [0mProjecting
     [32mband 4[0m
  File "/usr/local/lib/python2.7/dist-packages/landsat/image.py", line 262, in _write_to_file
     [32mband 3[0m
    band = self._color_correction(band, self.bands[i], 0, coverage)
     [32mband 2[0m
  File "/usr/local/lib/python2.7/dist-packages/landsat/image.py", line 276, in _color_correction
[34m===> [0mCalculating cloud and snow coverage from QA band
    p_low, cloud_cut_low = self._percent_cut(band, low, 100 - (coverage * 3 / 4))
     [32mcloud/snow coverage: 153.67[0m
  File "/usr/local/lib/python2.7/dist-packages/landsat/image.py", line 286, in _percent_cut
[34m===> [0mFinal Steps
    return numpy.percentile(color[numpy.logical_and(color > 0, color < 65535)], (low, high))
     [32mColor correcting band 4[0m
  File "/usr/lib/python2.7/dist-packages/numpy/lib/function_base.py", line 2820, in percentile
    return _compute_qth_percentile(sorted, q, axis, out)
  File "/usr/lib/python2.7/dist-packages/numpy/lib/function_base.py", line 2826, in _compute_qth_percentile
    for qi in q]
  File "/usr/lib/python2.7/dist-packages/numpy/lib/function_base.py", line 2835, in _compute_qth_percentile
    raise ValueError("percentile must be either in the range [0,100]")
ValueError: percentile must be either in the range [0,100]
```

**Comment by njwilson23** [comment]
It appears that cloud pixels are counted twice at https://github.com/developmentseed/landsat-util/blob/f0c229b9b7d466ff90c39ea69cf136db8d6130e8/landsat/image.py#L301.

Was this the intention?

**Comment by haoliangyu** [comment]
Another reason is at many situations (especially at the edge of objects) the distinction between cloud and snow, or among different types of cloud/snow , is ambiguous. A pixel could be identified as both cloud and snow with high confidence. The consequence is \# of snow pixels + \# of cloud pixels  > # of snow and cloud pixels. The current algorithm simply adds the pixel number of different cloud and snow types, so results in larger number than the reality.

**Comment by scisco** [comment]
Fixed in #183
